plugins/BEdita/API/src/Controller/ObjectsController.php

```diff
@@ -80,7 +80,7 @@ public function initialize()
 
         $type = $this->request->getParam('object_type', $this->request->getParam('controller'));
         try {
-            $this->objectType = TableRegistry::get('ObjectTypes')->get($type);
+            $this->objectType = TableRegistry::get('ObjectTypes')->get($type, ['denySingular' => true]);
```

Comment:
This way is ok, but in this case we don't want to only exclude singular forms, but also IDs and camelized forms.

I'd simply build the query manually this way:
```php
$this->objectType = TableRegistry::get('ObjectTypes')->find()
    ->where(['name' => $type])
    ->firstOrFail();
```
…but this does not benefits from caching. Another option would be:
```php
$this->objectType = TableRegistry::get('ObjectTypes')->get($type);
if ($type !== $this->objectType->name) {
    throw new NotFoundException(__d('bedita', 'Did you mean "{0}"?', $this->objectType->name));
    // or, if we feel like doing so:
    return $this->redirect(/* url rebuilt using correct type */, 304);
}
```

Reply:
I'd go with the second solution, simplifies PR too.
It provides a suggestion which is a good idea. I'd throw an error without redirect for now.

Reply:
I agree that throwing the error is a much better solution, at least for now.